core/src/test/scala/org/apache/spark/util/UtilsSuite.scala

```diff
@@ -1309,6 +1309,97 @@ class UtilsSuite extends SparkFunSuite with ResetSystemProperties with Logging {
     assert(Utils.buildLocationMetadata(paths, 15) == "[path0, path1, path2]")
     assert(Utils.buildLocationMetadata(paths, 25) == "[path0, path1, path2, path3]")
   }
+
+  test("checkHosts support IPV6 and IPV4") {
+    // IPV4 ips
+    Utils.checkHost("0.0.0.0")
+    intercept[AssertionError] {
+      Utils.checkHost("0.0.0.0:0")
+    }
```

Comment:
In this case, it would be great if we check the error message to prevent future regression.